airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/connectors/publish/pipeline.py

```diff
@@ -431,9 +478,39 @@ async def _run(self) -> StepResult:
         return await super()._run()
 
 
+class CheckConnectorVersionIncrement(Step):
+    context: PublishConnectorContext
+
+    @property
+    def title(self) -> str:
+        return "Check if connector version was incremented"
+
+    async def _run(self) -> StepResult:
+        # Check if running from manual GitHub workflow
+        if os.getenv("GITHUB_EVENT_NAME") == "workflow_dispatch":
+            return StepResult(
+                step=self,
+                status=StepStatus.SUCCESS,
+                stdout="Manual publish requested - overriding version increment check.",
+            )
+
+        # Check if version was incremented
+        if not self.context.connector.has_metadata_change:
```

Comment:
I would like you to implement a more precise check of whether the version has been incremented. I agree that the `metadata.dockerImageTag` is the best way to see if there is a new version.

Check out the implementation that already exists here: https://github.com/airbytehq/airbyte/blob/28035240d2fb030db0020e43fe7265b4a255dc73/airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/connectors/test/steps/common.py#L132